cisco_aci/datadog_checks/cisco_aci/api.py

```diff
@@ -60,11 +102,51 @@ def __init__(self, aci_urls, username, password, verify=False, timeout=10, log=N
         # This is used in testing
         self._refresh_sessions = True
 
+        self.password = password
+
+        self.cert_key_password = cert_key_password
+
+        self.cert_name = None
+        self.cert_key = None
+        if cert_name and cert_key:
+            self.cert_name = cert_name
+            self.cert_key = cert_key
+        elif not password:
+            msg = "You need to have either a password or a cert"
+            raise ConfigurationException(msg)
+
     def close(self):
         for session in self.sessions:
             session.close()
 
-    def login(self):
+    def setup_cert_login(self):
+        if self._refresh_sessions:
+            # ensure sessions are an empty array
+            self.sessions = []
+        for aci_url in self.aci_urls:
+            if not self._refresh_sessions:
+                for session_wrapper in self.sessions:
+                    if session_wrapper.aci_url == aci_url:
+                        session = session_wrapper.session
+                        break
+            else:
+                session = Session()
+
+            if self._refresh_sessions:
+                session_wrapper = SessionWrapper(aci_url, session,
+                                                 cert_name=self.cert_name,
+                                                 cert_key=self.cert_key,
+                                                 verify=self.verify,
+                                                 timeout=self.timeout,
+                                                 appcenter=self.appcenter,
+                                                 username=self.username,
+                                                 cert_key_password=self.cert_key_password,
+                                                 log=self.log)
+                self.sessions.append(session_wrapper)
+            else:
+                session_wrapper.apic_cookie = self.apic_cookie
```

Comment:
If `self.sessions` is ever empty `session_wrapper` will never be defined.

Reply:
yea that's a mistake, I removed it